Mastermind/Mastermind.cs

```diff
@@ -1,86 +1,132 @@
 
-            // leave this command at the end so your program does not close automatically
-            Console.ReadLine();
-        }
-        
-        public static bool CheckSolution(char[] guess)
+        public Ball (string letter)
         {
-            // Your code here
+            this.Letter = letter;
+        }
+    }
+    class Row
+    {
+        public Ball[] balls = new Ball[4];
+        //creates array of 4 of the Ball class
 
-            return false;
+        public Row (Ball[] balls)
+        {
+            this.balls = balls;
         }
-        
-        public static string GenerateHint(char[] guess)
+
+        public string Balls
+        {get
         {
-            // Your code here
-            return " ";
+            foreach(var ball in this.balls)
+            {
+                Console.WriteLine(ball.Letter);
+            }
+        return "";
         }
-        
-        public static void InsertCode(char[] guess)
+        }
+    }
+
+    class Game
+    {
+        public List<Row> rows = new List<Row>();
+        //creates a private list of the Row class
+        public string[] answer = new string[4];
+        //create a new array of strings with the size of 4 to hold our answer
+        public int guesses = 0;
+
+        public Game(string[] answer, int guesses)
         {
-            // Your code here
+            this.answer = answer;
+            this.guesses = guesses;
         }
-        
-        public static void CreateBoard()
+
+        public string Score (Row row)
         {
-            for (var i = 0; i < allowedAttempts; i++)
+            string[] answerClone = (string[]) this.answer.Clone ();
+            // red is correct letter and correct position
+            // white is correct letters minus red
+            // this.answer => ["a", "b", "c", "d"]
+            // row.balls => [{ Letter: "c" }, { Letter: "b" }, { Letter: "d" }, { Letter: "a" }]
+            int red = 0;
+            for (int i = 0; i<4; i++)
+            {
+                if (answerClone[i] == row.balls[i].Letter)
+                red++;
+            }
+            int white = 0;
+            for(int i = 0; i<4; i++)
             {
-                board[i] = new string[codeSize + 1];
-                for (var j = 0; j < codeSize + 1; j++)
+                int foundIndex = Array.IndexOf(answerClone, row.balls[i].Letter);
+                if(foundIndex > -1)
                 {
-                    board[i][j] = " ";
+                    white++;
+                    answerClone[foundIndex] = null;
                 }
             }
+            return $"{red} - {white - red}";
         }
-        
-        public static void DrawBoard()
+    }
+    class program
+    {
+        public static void Main(string[] args)
         {
-            for (var i = 0; i < board.Length; i++)
+            string[] answer = CreateAnswer();
+            
+            Game game = new Game(answer, 4);
```

Comment:
I would have expected the Game class to make use of the number of guesses passed in to tell you if the user has another turn or not.